llvm/lib/Target/AIE/AIEBaseInstrInfo.cpp

```diff
@@ -267,17 +276,20 @@ unsigned AIEBaseInstrInfo::insertBranch(
       *BytesAdded += getInstSizeInBytes(MI);
     return 1;
   }
+  assert(Cond.size() == 2);
 
-  if (Cond.size() == 3) {
-    BuildMI(&MBB, DL, get(Cond[0].getImm())).addMBB(TBB).add(Cond[2]);
-    return 1;
-  }
-
-  // Either a one or two-way conditional branch.
+  // Cond[0] contains the branch opcode (JZ, JNZ, PseudoLoopEnd)
+  // Cond[1] contains opcode dependent information, the last-bundle symbol for
+  // PseudoLoopEnd, the condition register for JZ/JNZ
   unsigned Opc = Cond[0].getImm();
-  MachineInstr &CondMI = *BuildMI(&MBB, DL, get(Opc)).add(Cond[1]).addMBB(TBB);
+  MachineInstrBuilder CBB = BuildMI(&MBB, DL, get(Opc));
```

Comment:
As we already have TBB, FBB as BB, CBB seems to be a BB as well because of the name. Maybe a different name is enough to state that this is an instr.